dotnet/src/SemanticKernel.Core/Functions/SKFunctionFromMethod.cs

```diff
@@ -22,65 +22,75 @@
 using Microsoft.SemanticKernel.Orchestration;
 
 #pragma warning disable IDE0130
-// ReSharper disable once CheckNamespace - Using the main namespace
+
 namespace Microsoft.SemanticKernel;
-#pragma warning restore IDE0130
 
 /// <summary>
-/// <see cref="ISKFunction"/> implementation backed by a delegate.
+/// Provides factory methods for creating <see cref="ISKFunction"/> instances backed by a .NET method.
 /// </summary>
 [DebuggerDisplay("{DebuggerDisplay,nq}")]
-internal sealed class NativeFunction : ISKFunction
+public sealed class KernelFunctionFromMethod : ISKFunction
```

Comment:
Is this the final name?

Reply:
@matthewbolanos, @markwallace-microsoft, what would you like to do here?  This is what we'd discussed, but then Mark suggested we instead consolidate down to having a single factory class per assembly, e.g. `SKFunction.CreateFunctionFromMethod`.  What would you prefer?

Reply:
We want to minimise the dependencies of SK.Core i.e. it shouldn't know anything about any YAML packages. 

So we would have:

**SK.Core**
```csharp
SKFunction.CreateFromPrompt(string prompt)
SKFunction.CreateFromPrompt(string prompt, PromptTemplateConfig promptTemplateConfig) // Support for current format
SKFunction.CreateFromPrompt(PromptModel promptModel)
SKFunction.CreateFromPromptJson(string json) // Deserialises json to an instance of PromptModel
SKFunction.CreateFromPromptJsonResource(string resourceName)
```

**SK.Functions.Yaml**
```csharp
SKFunctionYaml.CreateFromPromptYaml(string yaml) // Deserialises yaml to an instance of PromptModel
SKFunctionYaml.CreateFromPromptYamlResource(string resourceName)
```

**SK.Functions.Markdown**
```csharp
SKFunctionMarkdown.CreateFromPromptMarkdown(string markdown) // Creates PromptModel from code blocks in markdown
SKFunctionMarkdown.CreateFromPromptMarkdownResource(string resourceName)
```

If someone wants to support a new serialised format which could be json or xml they would do

**Contoso.Functions.Foo**
```csharp
SKFunctionFoo.CreateFromPromptFooJson(string json) // Deserialises custom json to an instance of PromptModel
SKFunctionFoo.CreateFromPromptFooXml(string xml) // Deserialises custom xml to an instance of PromptModel
```

Reply:
Added this to our chat and OK to merge and rename if needed later